TESTS/netsocket/gethostbyname/main.cpp

```diff
@@ -0,0 +1,106 @@
+#include "mbed.h"
+#include "greentea-client/test_env.h"
+#include "unity.h"
+#include "utest.h"
+#include MBED_CONF_APP_HEADER_FILE
+
+using namespace utest::v1;
+
+// Hostname for testing against
+// Must have A and AAAA records
+#ifndef MBED_DNS_TEST_HOST
+#define MBED_DNS_TEST_HOST "connector.mbed.com"
+#endif
+
+
+// Address info from stack
+const char *ip_literal;
+nsapi_version_t ip_pref;
+const char *ip_pref_repr;
+
+// Network setup
+NetworkInterface *net;
+
+void net_bringup() {
+    net = MBED_CONF_APP_OBJECT_CONSTRUCTION;
+    int err =  MBED_CONF_APP_CONNECT_STATEMENT;
+    TEST_ASSERT_EQUAL(0, err);
+    printf("MBED: Connected to network\n");
+    printf("MBED: IP Address: %s\n", net->get_ip_address());
+
+    ip_literal = net->get_ip_address();
+    ip_pref = SocketAddress(ip_literal).get_ip_version();
+    ip_pref_repr = (ip_pref == NSAPI_IPv4) ? "ipv4" :
+                   (ip_pref == NSAPI_IPv6) ? "ipv6" : "unspec";
+}
+
+
+// DNS tests
+void test_dns_query() {
+    SocketAddress addr;
+    int err = net->gethostbyname(MBED_DNS_TEST_HOST, &addr);
+    printf("DNS: query \"%s\" => \"%s\"\n",
+            MBED_DNS_TEST_HOST, addr.get_ip_address());
+
+    TEST_ASSERT_EQUAL(0, err);
+    TEST_ASSERT((bool)addr);
+    TEST_ASSERT(strlen(addr.get_ip_address()) > 1);
+}
+
+void test_dns_query_pref() {
+    SocketAddress addr;
+    int err = net->gethostbyname(MBED_DNS_TEST_HOST, &addr, ip_pref);
+    printf("DNS: query %s \"%s\" => \"%s\"\n",
+            ip_pref_repr, MBED_DNS_TEST_HOST, addr.get_ip_address());
+
+    TEST_ASSERT_EQUAL(0, err);
+    TEST_ASSERT((bool)addr);
+    TEST_ASSERT(strlen(addr.get_ip_address()) > 1);
+    TEST_ASSERT_EQUAL(ip_pref, addr.get_ip_version());
+}
+
+void test_dns_literal() {
+    SocketAddress addr;
+    int err = net->gethostbyname(ip_literal, &addr);
+    printf("DNS: literal \"%s\" => \"%s\"\n",
+            ip_literal, addr.get_ip_address());
+
+    TEST_ASSERT_EQUAL(0, err);
+    TEST_ASSERT((bool)addr);
+    TEST_ASSERT(strlen(addr.get_ip_address()) > 1);
+    TEST_ASSERT(strcmp(ip_literal, addr.get_ip_address()) == 0);
+}
+
+void test_dns_literal_pref() {
+    SocketAddress addr;
+    int err = net->gethostbyname(ip_literal, &addr, ip_pref);
+    printf("DNS: literal %s \"%s\" => \"%s\"\n",
```

Comment:
I know, anyway asking for your opinion or others. Are these prints valuable if test passes? if it passes, all should be OK. If it fails, than this debug info might be vital.